camel-idea-plugin/src/main/java/org/apache/camel/idea/gutter/CamelRouteLineMarkerProvider.java

```diff
@@ -69,4 +93,116 @@ private CamelPreferenceService getCamelPreferenceService() {
+        }
+
+        return null;
+    }
+
+    /**
+     * Searches in the project all the route destinations for the given {@link PsiElement}.
+     * Example for Java routes: for 'from("file:inbox")' returns all elements that matches 'to("file:inbox")'
+     * <p>
+     * Since Intellij API supports only searches with one keyword the search is made using just the Camel component name and then further refined.
+     * </p>
+     *
+     * @param startElement the {@link PsiElement} that contains the definition for a route start
+     * @return a list of {@link PsiElement} with all the route ends.
+     */
+    private List<PsiElement> findRouteDestinationForPsiElement(PsiElement startElement) {
+        List<PsiElement> psiElements = new ArrayList<>();
+        String route = findRouteFromElement(startElement);
+
+        if (route == null || route.isEmpty()) {
+            return psiElements;
+        }
+        PsiSearchHelper helper = PsiSearchHelper.SERVICE.getInstance(startElement.getProject());
+        //get the component name and search only using that
+        String componentName = route.split(":")[0];
+
+        helper.processElementsWithWord((psiElement, offsetInElement) -> {
+            if (psiElement instanceof XmlToken) {
+                PsiElement xmlElement = findXMLElement(route, (XmlToken) psiElement);
+                if (xmlElement != null) {
+                    psiElements.add(xmlElement);
+                }
+            }
+            if (psiElement instanceof PsiLiteralExpression) {
+                PsiElement javaElement = findJavaElement(route, (PsiLiteralExpression) psiElement);
+                if (javaElement != null) {
+                    psiElements.add(javaElement);
+                }
+            }
+            return true;
+        }, getSearchScope(startElement.getProject()), componentName, UsageSearchContext.ANY, false);
+
+        return psiElements;
+    }
+
+    /**
+     * Further refine search in order to match the exact Java Camel route.
+     * Checks if the given {@link PsiElement} contains a 'to' method that points to the give route.
+     *
+     * @param route      the complete Camel route to search for
+     * @param psiElement the {@link PsiLiteralExpression} that might contain the complete route definition
+     * @return the {@link PsiElement} that contains the exact match of the Camel route, null if there is no exact match
+     */
+    private PsiElement findJavaElement(String route, PsiLiteralExpression psiElement) {
+        if (route.equals(psiElement.getValue())) {
+            //the method 'to' is a PsiIdentifier not a PsiMethodCallExpression because it's part of method invocation chain
+            PsiMethodCallExpression methodCall = PsiTreeUtil.getParentOfType(psiElement, PsiMethodCallExpression.class);
+            if (methodCall != null && "to".equals(methodCall.getMethodExpression().getReferenceName())) {
```

Comment:
You can also have `toD` or `toF` in Camel. The former is used more than the latter which is a shortcut for using String.format. to build the uri string.